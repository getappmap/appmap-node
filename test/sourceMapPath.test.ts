import { integrationTest, readAppmap, runAppmapNode } from "./helpers";

integrationTest("mapping a bundled script", () => {
  expect(runAppmapNode("./built/index.js").status).toBe(0);
  expect(readAppmap()).toMatchSnapshot();
});
