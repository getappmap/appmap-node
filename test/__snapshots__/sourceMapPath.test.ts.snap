// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapping a bundled script 1`] = `
{
  "classMap": [
    {
      "children": [
        {
          "children": [
            {
              "location": "original.mjs:3",
              "name": "foo",
              "static": true,
              "type": "function",
            },
          ],
          "name": "original",
          "type": "class",
        },
      ],
      "name": "appmap-node",
      "type": "package",
    },
  ],
  "events": [
    {
      "defined_class": "original",
      "event": "call",
      "id": 1,
      "lineno": 3,
      "method_id": "foo",
      "parameters": [
        {
          "class": "String",
          "name": "msg",
          "value": "'hello'",
        },
      ],
      "path": "original.mjs",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 2,
      "parent_id": 1,
      "thread_id": 0,
    },
  ],
  "metadata": {
    "app": "appmap-node",
    "client": {
      "name": "appmap-node",
      "url": "https://github.com/getappmap/appmap-node",
      "version": "test node-appmap version",
    },
    "language": {
      "engine": "Node.js",
      "name": "javascript",
      "version": "test node version",
    },
    "name": "test process recording",
    "recorder": {
      "name": "process",
      "type": "process",
    },
  },
  "version": "1.12",
}
`;
